### PR TITLE
chore: references are now parsed in the JSON Schema processor

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR solves the issue of parsing references before the simplification stage.

### CODE REVIEW

File: src/processors/JsonSchemaInputProcessor.ts [hunk 1]
```diff
@@ -47,13 +49,21 @@ export class JsonSchemaInputProcessor extends AbstractInputProcessor {
 
     /**
      * Process a draft 7 schema
+     * 
      * @param input to process as draft 7
      */
-    private processDraft7(input: any) : CommonInputModel {
-        const schema = Schema.toSchema(input);
+    private async processDraft7(input: any) : Promise<CommonInputModel> {
+        const refParser = new $RefParser;
```
Comment (magicmatatjahu) [inline src/processors/JsonSchemaInputProcessor.ts:56]: Is new parser needed for every parsing?

Comment (jonaslagoni) [reply src/processors/JsonSchemaInputProcessor.ts:56]: Ehhh, goood question. I guess not? 😄

Comment (jonaslagoni) [reply src/processors/JsonSchemaInputProcessor.ts:56]: let me try

Comment (magicmatatjahu) [reply src/processors/JsonSchemaInputProcessor.ts:56]: Ahh, it should be new 😄 I remember one issue in parser-js that some state in refParser was reassigned in asynchronus functions, so we should stay with it.

**Review by magicmatatjahu — APPROVED** [review]

**Pull request merged**